src/edu/ucsb/nceas/metacat/properties/PropertiesWrapper.java

```diff
@@ -409,6 +443,50 @@ private void initialize() throws GeneralPropertyException {
         }
     }
 
+    /**
+     * From the property 'application.envSecretKeys', retrieve the colon-delimited mappings
+     * between: camelCase / period-delimited properties keys (as found elsewhere in the
+     * properties files), and SCREAMING_SNAKE_CASE environment variable keys. Then use these to
+     * populate the envSecretKeyMappings Properties, used elsewhere to check for overrides when
+     * getProperty() is called.
+     * The property file entry should be in the form:
+     * <code>
+     *   application.secretKeys=         \
+     *     :some.CamelCase=METACAT_VAR_A \
+     *     :another.key=METACAT_VAR_B    \
+     *     (...etc.)
+     * </code>
+     */
+    private void initializeSecretsFromEnvVars() {
+
+        String secretMappingsStr = mainProperties.getProperty("application.envSecretKeys");
+        final String ENTRY_DELIMITER = ":";
+        final String KEY_VAL_DELIMITER = "=";
+        if (secretMappingsStr != null && !secretMappingsStr.isEmpty()) {
+            String[] mappingList = secretMappingsStr.split(ENTRY_DELIMITER);
+            for (String eachMapping : mappingList) {
+                String[] keyValPair = eachMapping.split(KEY_VAL_DELIMITER);
+                String key = keyValPair[0].trim();
+                String value = keyValPair[1].trim();
+                if (key.isEmpty() || value.isEmpty()) {
+                    logMetacat.debug(
```

Comment:
May we increase the debug level to error, which is the default debug level in Metacat? So operators can find the issue on the log file easily.

Reply:
good idea - will do!

Reply:
(agreed in slack that WARN was more appropriate, since it's not truly an error condition)